chat/public/js/src/chat.js

```diff
@@ -488,6 +488,24 @@ function ChatXBlock(runtime, element, init_data) {
     };
 
     /**
+     * isFinalStep: returns true if current step doesn't exist or has no responses.
+       This is the JS equivalent of the _is_final_step method of the XBlock.
+     */
+    var isFinalStep = function(step) {
+        var steps_dict = init_data["steps"];
+        // Step with this ID does not exists, which means the chat is complete.
+        if (!(step in steps_dict)) {
+            return true;
+        }
+        // Step exists, but has no user responses available, which means this is the final step.
+        if (!(steps_dict[step].responses)) {
```

Comment:
In javascript, empty array is not falsey, so I think you need to check whether its length is zero.